core/appeal/service.go

```diff
@@ -1121,6 +1129,131 @@ func getPolicy(a *domain.Appeal, p *domain.Provider, policiesMap map[string]map[
 	return policy, nil
 }
 
+func (s *Service) populateAppealMetadata(ctx context.Context, a *domain.Appeal, p *domain.Policy) error {
+	if p.AppealMetadataSources == nil {
+		return nil
+	}
+
+	eg, _ := errgroup.WithContext(ctx)
+
+	appealMetadata := map[string]interface{}{}
+	for key, metadata := range p.AppealMetadataSources {
+		key, metadata := key, metadata
+		eg.Go(func() error {
+			switch metadata.Type {
+			case "http":
+				if metadata.Config.URL == "" {
+					return fmt.Errorf("URL cannot be empty for http type")
+				}
+
+				metadataCl, err := s.getMetadataClient(metadata.Config)
+				if err != nil {
+					return fmt.Errorf("key: %s, %w", key, err)
+				}
+
+				res, err := metadataCl.MakeRequest()
+				if err != nil || (res.StatusCode < 200 && res.StatusCode > 300) {
+					if !metadata.Config.AllowFailed {
+						return fmt.Errorf("error fetching resource: %w", err)
+					}
+				}
+
+				var responseBody interface{}
+				err = json.NewDecoder(res.Body).Decode(responseBody)
+				if err != nil {
+					return fmt.Errorf("error decoding response: %w", err)
+				}
+
+				responseMap := responseBody.(map[string]interface{})
+				params := map[string]interface{}{
+					"response": responseMap,
+					"appeal":   a,
+				}
+				value, err := s.getMetadataValue(metadata.Type, key, metadata.Value, params)
+				if err != nil {
+					return fmt.Errorf("error parsing value: %w", err)
+				}
+				appealMetadata[key] = value
```

Comment:
need to define mutex outside the loop and use it here to avoid datarace
```suggestion
				mu.Lock()
				appealMetadata[key] = value
				mu.Unlock()
```